apps/docs/components/ProjectConfigVariables/ProjectConfigVariables.ComboBox.tsx

```diff
@@ -0,0 +1,103 @@
+import { Check, ChevronsUpDown } from 'lucide-react'
+import { noop } from 'lodash'
+import { useState } from 'react'
+import {
+  Button_Shadcn_ as Button,
+  Popover_Shadcn_ as Popover,
+  PopoverTrigger_Shadcn_ as PopoverTrigger,
+  PopoverContent_Shadcn_ as PopoverContent,
+  Command_Shadcn_ as Command,
+  CommandInput_Shadcn_ as CommandInput,
+  CommandEmpty_Shadcn_ as CommandEmpty,
+  CommandItem_Shadcn_ as CommandItem,
+  CommandGroup_Shadcn_ as CommandGroup,
+  cn,
+  ScrollArea,
+} from 'ui'
+
+export interface ComboBoxOption {
+  id: string
+  value: string
+  displayName: string
+}
+
+export function ComboBox<Opt extends ComboBoxOption>({
+  isLoading,
+  disabled,
+  name,
+  options,
+  selectedOption,
+  onSelectOption = noop,
+  className,
+}: {
+  isLoading: boolean
+  disabled?: boolean
+  name: string
+  options: Opt[]
+  selectedOption?: string
+  onSelectOption?: (newValue: string) => void
+  className?: string
+}) {
+  const [open, setOpen] = useState(false)
+
+  const selectedOptionDisplayName = options.find(
+    (option) => option.value === selectedOption
+  )?.displayName
+
+  return (
+    <Popover open={open} onOpenChange={setOpen}>
+      <div className={className}>
+        <PopoverTrigger asChild aria-label={`Select ${name}`} disabled={disabled}>
```

Comment:
you could move `aria-label={`Select ${name}`} disabled={disabled}` to the Button below to avoid any confusion to anyone later maybe?